### PR TITLE
V3.0.0 updates

### PR DESCRIPTION
**Pull request by caesarxuchao**
I'll create the v3.0.0 release after this is merged.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,15 @@ fixing this.
 
 # HEAD (changes that will go into the next release)
 
+# v3.0.0
+
+Bug fixes:
+* Use OS-specific libs when computing client User-Agent in kubectl, etc. (https://github.com/kubernetes/kubernetes/pull/44423)
+* kubectl commands run inside a pod using a kubeconfig file now use the namespace specified in the kubeconfig file, instead of using the pod namespace. If no kubeconfig file is used, or the kubeconfig does not specify a namespace, the pod namespace is still used as a fallback. (https://github.com/kubernetes/kubernetes/pull/44570)
+* Restored the ability of kubectl running inside a pod to consume resource files specifying a different namespace than the one the pod is running in. (https://github.com/kubernetes/kubernetes/pull/44862)
+* Fix a bug that caused invalid Tolerations to be applied to Pods created by DaemonSets. (https://github.com/kubernetes/kubernetes/pull/45349)
```
Comment (lavalamp) [inline CHANGELOG.md]: Is this really a client fix?

File: README.md [hunk 2]
```diff
@@ -80,13 +80,13 @@ We will backport bugfixes--but not new features--into older versions of
...
-| client-go 1.5       | +              | +              | -              | -              |
-| client-go 2.0       | +              | +              | ✓              | -              |
-| client-go 3.0 beta  | +              | +              | +              | ✓              |
-| client-go HEAD      | +              | +              | +              | +              |
+|                     | Kubernetes 1.3 | Kubernetes 1.4 | Kubernetes 1.5 | Kubernetes 1.6 | Kubernetes 1.7 |
+|---------------------|----------------|----------------|----------------|----------------|----------------|
+| client-go 1.4       | +              | ✓              | -              | -              | -              |
+| client-go 1.5       | +              | +              | -              | -              | -              |
+| client-go 2.0       | +              | +              | ✓              | -              | -              |
+| client-go 3.0       | + †            | +              | +              | ✓              | -              |
```
Comment (lavalamp) [inline README.md]: @caesarxuchao I pushed some changes to your branch -- take a look and see if this still represents what you were trying to communicate.

File: CHANGELOG.md (hunk 1 above)
Comment (caesarxuchao) [inline CHANGELOG.md]: Removed. The changes affect client-go/pkg/api/v1/helpers.go, but don't really affect the correctness of client-go.

File: README.md (hunk 2 above)
Comment (caesarxuchao) [inline README.md]: Do you still have your branch? I force pushed the branch..

File: README.md [hunk 3]
```diff
@@ -80,20 +80,24 @@ We will backport bugfixes--but not new features--into older versions of
...
 
 Key:
 
 * `✓` Exactly the same features / API objects in both client-go and the Kubernetes
   version.
 * `+` client-go has features or api objects that may not be present in the
-  Kubernetes cluster, but everything they have in common will work.
+  Kubernetes cluster, but everything they have in common will work. Please
+  note that alpha APIs may vanish or change significantly in a single release.
+* `†` client-go has new features or api objects, and some APIs running in the
```
Comment (caesarxuchao) [inline README.md]: I pushed another commit that changes several "+" to "†". In the 1.5 and 1.6 releases, client-go removed support for some alpha features.

**Comment by caesarxuchao** [comment]
@lavalamp PTAL.

File: README.md (hunk 3 above)
Comment (lavalamp) [inline README.md]: Note that I modified the '+' one to cover alpha APIs. I don't think we need a special symbol for alpha APIs.

Comment (caesarxuchao) [inline README.md]: I see. Basically we are saying we won't bump major version of client-go when we remove the support for alpha API.

I saw you used `†` three times, what non-alpha API support was deprecated in those cases?

Comment (lavalamp) [inline README.md]: There may not be a thing we actually deprecated, just it is possible that such a thing may exist.

File: README.md [hunk 4]
```diff
@@ -80,20 +80,24 @@ We will backport bugfixes--but not new features--into older versions of
...
-| client-go 1.4       | +              | ✓              | -              | -              |
-| client-go 1.5       | +              | +              | -              | -              |
-| client-go 2.0       | +              | +              | ✓              | -              |
-| client-go 3.0 beta  | +              | +              | +              | ✓              |
-| client-go HEAD      | +              | +              | +              | +              |
+|                     | Kubernetes 1.3 | Kubernetes 1.4 | Kubernetes 1.5 | Kubernetes 1.6 | Kubernetes 1.7 |
+|---------------------|----------------|----------------|----------------|----------------|----------------|
+| client-go 1.4       | +              | ✓              | -              | -              | -              |
+| client-go 1.5       | +              | +              | -              | -              | -              |
+| client-go 2.0       | †              | †              | ✓              | -              | -              |
```
Comment (lavalamp) [inline README.md]: I think there should always be a release that is a + between the check and the dagger marked releases, or we are violating a deprecation policy?

Comment (caesarxuchao) [inline README.md]: I agree with the statement, but i I think that's guaranteed by the version skew policy rather than the deprecation policy.

Comment (caesarxuchao) [inline README.md]: I removed the last commit, so the table looks like what you last edited. I'll merge it as is.

**Pull request merged**